### DISCUSSION TITLE
How to be sure when the model is fully instantiated in scene?

### DISCUSSION THREAD

**Discussion by massimilianogaio** [atteneder/glTFast#591] [Q&A] [answered]
HI. I currently use the default `GLTFast.GltfAsset` script attached to a gameObject to load models into the scene.
In order to access the instantiated model meshes, I wait for `GltfAssetBase.IsDone == true`. Using this control though, the object is not yet instantiated in scene and I still can't access the created gameObject. So I have to add also a `yield return new WaitForEndOfFrame();`
This is the code:
```
protected IEnumerator LoadModelAsync(string uri)
{
    var gltf = gameObject.GetComponent<GLTFast.GltfAsset>();
    gltf.Url = uri;
    yield return (new WaitUntil (() => gltf.IsDone));
    yield return new WaitForEndOfFrame();

    OnImportFinished(transform.GetChild(0).gameObject);
}
```

For most models, this works. But if the model to be imported has many children (>>200), it is not instantiated in that single frame. So the the code above doesn't work anymore because transform.GetChild(0) is not `null` but the children are not in the scene yet. 
My question is: is there a better approach to determine when the model is instantiated in scene? Thank you.

**Comment by WoodsFiend** [comment, marked as answer]
You should go async instead of IEnumerator

```
    private async Task<bool> DownloadModelAsync(string uri)
    {
        var didLoad = await gltf.Load(uri);
        var childCount = transform.childCount;
        
        //This handles the case of already having some children before loading model
        if (childCount > initialChildCount)
        {
            GameObject modelRoot = transform.GetChild(childCount).gameObject;
            OnImportFinished(modelRoot);
        }
        return didLoad;
    }
```

**Reply by massimilianogaio** [reply]
Thank you. Yes, I figured out to use the await function few weeks ago.
